src/commands/SinceCommand.ts

```diff
@@ -209,46 +224,86 @@ async function execSinceCommandAux(destinationRoomId: string, event: any, mjolni
-                    return {
-                        html: text_,
-                        text: text_,
-                    }
+                    // Store list of affected users, in case we need it later.
+                    /* no need to await */ mjolnir.client.sendStateEvent(targetRoomId, "org.matrix.mjolnir.since.kick", "", results);
+
+                    return formatResult("kick", targetRoomId, joins, results);
                 }
                 case Action.Ban: {
-                    const joins = mjolnir.roomJoins.getUsersInRoom(targetRoomId, minDate, maxEntries);
-
-                    let results = { good: 0, bad: 0};
                     for (let join of joins) {
                         try {
                             await mjolnir.client.banUser(join.userId, targetRoomId, reason);
-                            results.good += 1;
+                            results.succeeded.push(join.userId);
                         } catch (ex) {
                             LogService.warn("SinceCommand", "Error while attempting to ban user", ex);
-                            results.bad += 1;
+                            results.failed.push(join.userId);
+                        }
+                    }
+
+                    // Store list of affected users, in case we need it later.
+                    /* no need to await */ mjolnir.client.sendStateEvent(targetRoomId, "org.matrix.mjolnir.since.ban", "", results);
+                    return formatResult("ban", targetRoomId, joins, results);
+                }
+                case Action.Mute: {
+                    const powerLevels = await mjolnir.client.getRoomStateEvent(targetRoomId, "m.room.power_levels", "") as {users: Record</* userId */ string, number>};
+
+                    for (let join of joins) {
+                        powerLevels.users[join.userId] = -1;
+                    }
+                    try {
+                        await mjolnir.client.sendStateEvent(targetRoomId, "m.room.power_levels", "", powerLevels);
+                        for (let join of joins) {
+                            results.succeeded.push(join.userId);
+                        }
+                    } catch (ex) {
+                        LogService.warn("SinceCommand", "Error while attempting to mute users", ex);
+                        for (let join of joins) {
+                            results.failed.push(join.userId);
                         }
                     }
-                    const text_ = `Attempted to ban ${joins.length} users from room ${targetRoomId}, ${results.good} kicked, ${results.bad} failures`;
-                    return {
-                        html: text_,
-                        text: text_
+
+                    // Store list of affected users, in case we need it later.
+                    /* no need to await */ mjolnir.client.sendStateEvent(targetRoomId, "org.matrix.mjolnir.since.mute", "", results);
```

Comment:
if we are reusing joins to unmute users the same way they were muted, do we need the mute/unmute state events anymore? When I suggested this I was thinking unmute would use this state event when removing the power levels too, but that might not make sense now

Reply:
I think that we can avoid it for now and add the state event only later, if we realize we need it.